terminal.md

```diff
@@ -0,0 +1,505 @@
+
+By contrast, a  **relative path**  starts from some given working directory, avoiding the need to provide the full absolute path. A filename can be considered as a relative path based at the current working directory. If the working directory is not the file's parent directory, a file not found error will result if the file is addressed by its name.
+>`../docs` is a relative path
+
+## <a name="command-basics"></a> Basics of Linux/Unix Commands
+
+### <a name="command-structure"></a> Command Structure
+The components of the command line are:
+-   the command
+-   any options required by the command
+-   the command's arguments (if required)
+
+The general form of a UNIX command is:
+**command [-option(s)] [argument(s)]**
+```bash
+#only the command
+clear
+
+#command with option
+ls -l
+
+#command with argument
+man pwd
+
+#command with option and argument
+rm -r project
+```
+### <a name="command-help"></a> Command Help
+-  `man <command>` is used to display the user manual of any command that we can run on the terminal. It provides a detailed view of the command. eg: `man ls`.
+- `help` command provides documentation for builtin commands. 
+```bash
+$ help pwd
+pwd: pwd [-LP]
+    Print the name of the current working directory.
+    
+   Options:
+     -L	print the value of $PWD if it names the current working directory
+     -P	print the physical directory, without any symbolic links
+   
+   By default, `pwd' behaves as if `-L' were specified.
+   
+   Exit Status:
+   Returns 0 unless an invalid option is given or the current directory
+   cannot be read.
+```
+## <a name="command-list"></a> List of Commands
+  
+### Print name of cwd (current working directory)
+Use `pwd` to print name of current/working directory
+```bash
+$ pwd
+/home/username
+```
+
+### Clear screen & command history
+Use `clear` or <kbd>Ctrl</kbd> + <kbd>L</kbd> to clear the screen of the cmd.
+  
+
+### Changing directories
+```bash
+$ pwd
+/home/username
+
+$ # providing an absolute path as argument
+$ cd /etc
+$ pwd
+/etc
+
+$ # to go back to previous working directory
+$ # if there's a directory named '-', use './-' to go that directory
+$ cd -
+/home/username
+$ pwd
+/home/username
+```
+-   Relative paths are well, relative to current working directory
+-   `.`  refers to current directory
+-   `..`  refers to directory one hierarchy above
+-   `../..`  refers to directory two hierarchies above and so on
+	```bash
+	$ pwd
+	/home/username
+
+	$ # go to directory one hierarchy above
+	$ cd ..
+	$ pwd
+	/home
+
+	$ # go to directory 'username' present in current directory
+	$ # './' is optional in this case
+	$ cd ./username
+	$ pwd
+	/home/username
+
+	$ # go to directory two hierarchies above
+	$ cd ../..
+	$ pwd
+	/
+	```
+-   `cd ~/`  or  `cd ~`  or  `cd`  will go to directory specified by  `HOME`  shell variable (which is usually set to user's home directory)
+	```bash
+	$ pwd
+	/
+	$ echo "$HOME"
+	/home/username
+
+	$ cd
+	$ pwd
+	/home/username
+	```
```

Comment:
The correct way to make code blocks and other elements part of a list item is to indent them with an equal number of spaces (2 in this case), not tabs.

Please make the changes everywhere else.

Eg:
```suggestion
  ```bash
  $ pwd
  /
  $ echo "$HOME"
  /home/username
  $ cd
  $ pwd
  /home/username
  ```
```

Reply:
The indentation here is not for creating a code block. (``` already does that) It's just for aligning the code block within the bullet point.